File: NX1899_NX1903_NX1907_NX1911_NX1915_NX1919/UGOPEN/SampleNXOpenApplications/.NET/NXOpenExamples/EX_Curve_CreateArc.cs
/*=============================================================================

                    Copyright (c) 2004 UGS PLM Solutions
                    Unpublished - All rights reserved

===============================================================================
File description: Example for Wrapped UFUNC (NXOpen.NET)
/*  */

using System;
using System.IO;
using NXOpen;
using NXOpen.UF;

namespace NetExample
{
    /// This example will create EX_Curve_CreateArc.prt, EX_Curve_CreateArc.log files 
    /// and creates an Arc using theUfSession.Curve.CreateArc()method.
    /// The program prints the report to the log file called "EX_Curve_CreateArc.log".
    public class EX_Curve_CreateArc
    {
        private static FileStream fs; 
        private static StreamWriter w;
        private static UFSession theUfSession;
        private static Session theSession;

        public int Execute()
        {
            Tag UFPart;
            string part_name = "EX_Curve_CreateArc";
            int units =2; 
            string name;
          
            theUfSession.Part.New(part_name, units, out UFPart);
            theUfSession.Part.AskPartName(UFPart, out name);
            w.WriteLine("Loaded: " + name);
            
            Tag arc, wcs;
            UFCurve.Arc arc_coords = new UFCurve.Arc();     
            
            /* Fill out the data structure */
            arc_coords.start_angle = 0.0;
            arc_coords.end_angle = 3.0;
            arc_coords.arc_center=new double[3];
            arc_coords.arc_center[0] = 0.0;
            arc_coords.arc_center[1] = 0.0;
            arc_coords.arc_center[2] = 1.0;
            arc_coords.radius = 2.0;
            
            theUfSession.Csys.AskWcs(out wcs);
            theUfSession.Csys.AskMatrixOfObject(wcs,out arc_coords.matrix_tag);
            theUfSession.Curve.CreateArc(ref arc_coords,out arc);
            theUfSession.Part.Save();
            return 0;
        }
        public static void Main(string[] args)
        {
            theSession=Session.GetSession();
            theUfSession= UFSession.GetUFSession();

            fs = new FileStream("EX_Curve_CreateArc.log", FileMode.Create, FileAccess.Write);
            w = new StreamWriter(fs); // create a stream writer 
            w.Write("Log Entry : \r\n");
            w.WriteLine("--Log entry goes here--"); 
            w.Flush(); // update underlying file

            if ( File.Exists("EX_Curve_CreateArc.prt") ) 
            {
                w.WriteLine("Remove EX_Curve_CreateArc.prt file from <Project Folder>\\bin\\Debug !!");
                w.WriteLine("EX_Curve_CreateArc.prt already exists. !!");
                w.Close();
                return;
            }

            try
            {
                EX_Curve_CreateArc curveTest1 = new EX_Curve_CreateArc();
                if (curveTest1.Execute()==0)
                {
                    w.WriteLine("Successful");
                }
                else
                {
                    w.WriteLine("Failed");
                }
            }
            catch(NXException e)
            {   
                w.WriteLine("Exception is: {0}", e.Message);
            }
            w.WriteLine("End of Log File");
            w.Close();
        }
        public static int GetUnloadOption(string dummy) 
        {
            return UFConstants.UF_UNLOAD_IMMEDIATELY;
        }
    }
}

File: NX1980_NX1984_NX1988_NX1992_NX1996_NX2000/UGOPEN/SampleNXOpenApplications/.NET/NXOpenExamples/EX_Curve_CreateArc.vb
' ==================================================================================================
'
'            Copyright (c) 2004 UGS PLM Solutions.
'                   Unpublished - All rights reserved
'
'====================================================================================================
'File description:
'Example to show how to create Arc using UF .NET Wrappers: uf_curve_CreateArc
' 
Option Strict Off
Imports System
Imports System.IO
Imports NXOpen
Imports NXOpen.UF
Imports NXOpen.Utilities



Module EX_Curve_CreateArc

    Dim testName As String = "EX_Curve_CreateArc"

    Dim theSession As Session
    Dim theUfSession As UFSession

    Sub Main()

        theSession = Session.GetSession()
        theUfSession = UFSession.GetUFSession()
        Dim stream As StreamWriter = New StreamWriter(testName + ".log")
        Dim prtFileName As String = New String(testName + ".prt")
        Dim prtFile As FileInfo = New FileInfo(prtFileName)
        prtFile = New FileInfo(prtFileName)
        stream.WriteLine("Log Entry : ")
        stream.WriteLine("--Log entry goes here--")

        If prtFile.Exists() Then
            stream.WriteLine("Remove " + prtFileName + " EX_Curve_CreateArc.prt file from <Project Folder> !!")
            stream.WriteLine(prtFileName + " already exists. !!")
            stream.WriteLine("End of Log File")
            stream.Close()
            Return
        End If

        Dim nError As Integer = Execute(stream, theUfSession)
        stream.WriteLine("End of Log File")
        stream.Close()

    End Sub

    Function Execute(ByVal stream As StreamWriter, ByVal theUfSession As UFSession) As Integer
        Dim PartTag As Tag
        Dim units As Integer = UFConstants.ENGLISH
        Dim PartName As String

        theUfSession.Part.[New](testName, units, PartTag)
        theUfSession.Part.AskPartName(PartTag, PartName)
        stream.WriteLine("New Part File Created :- " + PartName)

        Dim arc, wcs As Tag
        Dim arc_coords As UFCurve.Arc = New UFCurve.Arc

        'Fill out the data structure */
        arc_coords.start_angle = 0.0
        arc_coords.end_angle = 3.0
        Dim array() As Double = {0.0, 0.0, 1.0}
        arc_coords.arc_center = array
        'arc_coords.arc_center(0) = 0.0
        'arc_coords.arc_center(1) = 0.0
        'arc_coords.arc_center(2) = 1.0
        arc_coords.radius = 2.0

        stream.WriteLine("Creating Arc:-")
        theUfSession.Csys.AskWcs(wcs)
        theUfSession.Csys.AskMatrixOfObject(wcs, arc_coords.matrix_tag)
        theUfSession.Curve.CreateArc(arc_coords, arc)
        stream.WriteLine("Successful")

        stream.WriteLine("Querying Arc:-")
        Dim created_arc_coords As UFCurve.Arc = New UFCurve.Arc
        theUfSession.Curve.AskArcData(arc, created_arc_coords)
        stream.Write("arc_center= {")
        stream.Write(created_arc_coords.arc_center(0).ToString())
        stream.Write(",")
        stream.Write(created_arc_coords.arc_center(1).ToString())
        stream.Write(",")
        stream.Write(created_arc_coords.arc_center(2).ToString())
        stream.WriteLine("}")

        stream.WriteLine("start_angle= " + created_arc_coords.start_angle.ToString)
        stream.WriteLine("end_angle= " + created_arc_coords.end_angle.ToString)
        stream.WriteLine("radius= " + created_arc_coords.radius.ToString)

        'Save the Part
        theUfSession.Part.Save()
        Return 0

    End Function

    Public Function GetUnloadOption(ByVal dummy As String) As Integer
        GetUnloadOption = UFConstants.UF_UNLOAD_IMMEDIATELY
    End Function

End Module
